src/compiler/commandLineParser.ts

```diff
@@ -710,13 +714,259 @@ namespace ts {
+            type: "list",
+            element: {
+                name: "exclude",
+                type: "string"
+            }
+        },
+        compileOnSaveCommandLineOption
+    ];
+
+    function commandLineOptionsToMap(options: CommandLineOption[]) {
+        return arrayToMap(options, option => option.name);
+    }
+
+    let _tsconfigRootOptionsMap: Map<CommandLineOption>;
+    function getTsconfigRootOptionsMap() {
+        if (_tsconfigRootOptionsMap === undefined) {
+            _tsconfigRootOptionsMap = commandLineOptionsToMap(tsconfigRootOptions);
+        }
+        return _tsconfigRootOptionsMap;
+    }
+
+    interface JsonConversionNotifier {
+        /** Notifies options object is being set with the optionKey and optionValue is being set */
+        onSetOptionKeyValue(optionsObject: string, option: CommandLineOption, value: CompilerOptionsValue): void;
+        /** Notify when root key value is being set */
+        onRootKeyValue(key: string, propertyName: PropertyName, value: CompilerOptionsValue, node: Expression): void;
+    }
+
+    /**
+     * Convert the json syntax tree into the json value
+     * @param jsonNode
+     * @param errors
+     */
+    export function convertToJson(sourceFile: JsonSourceFile, errors: Diagnostic[]): any {
+        return convertToJsonWorker(sourceFile, errors);
+    }
+
+    /**
+     * Convert the json syntax tree into the json value
+     * @param jsonNode
+     * @param errors
+     */
+    function convertToJsonWorker(sourceFile: JsonSourceFile, errors: Diagnostic[], knownRootOptions?: Map<CommandLineOption>, optionsIterator?: JsonConversionNotifier): any {
+        if (!sourceFile.jsonObject) {
+            if (sourceFile.endOfFileToken) {
+                return {};
+            }
+            return undefined;
+        }
+
+        return convertObjectLiteralExpressionToJson(sourceFile.jsonObject, knownRootOptions);
+
+        function convertObjectLiteralExpressionToJson(node: ObjectLiteralExpression, options?: Map<CommandLineOption>, extraKeyDiagnosticMessage?: DiagnosticMessage, optionsObject?: string): any {
+            const result: any = {};
+            for (const element of node.properties) {
+                if (element.kind !== SyntaxKind.PropertyAssignment) {
+                    errors.push(createDiagnosticForNodeInSourceFile(sourceFile, element, Diagnostics.Property_assignment_expected));
+                    continue;
+                }
+
+                if (element.questionToken) {
+                    errors.push(createDiagnosticForNodeInSourceFile(sourceFile, element.questionToken, Diagnostics._0_can_only_be_used_in_a_ts_file, "?"));
+                }
+                if (!isDoubleQuotedString(element.name)) {
+                    errors.push(createDiagnosticForNodeInSourceFile(sourceFile, element.name, Diagnostics.String_literal_with_double_quotes_expected));
+                }
+
+                const keyText = getTextOfPropertyName(element.name);

```

Comment:
https://github.com/Microsoft/TypeScript/pull/12336/files#diff-057bf8560e435e388ba45f7810c3d08dR844 already reported error for this isn't it.